lib/assert.js

```diff
@@ -95,43 +93,22 @@ function innerFail(obj) {
 }
 
 /**
- * @param {any} actual
- * @param {any} expected
- * @param {string | Error} [message]
- * @param {string} [operator]
- * @param {Function} [stackStartFn]
+ * Throws an AssertionError with the given message.
+ * @param {any | Error} [message]
  */
-function fail(actual, expected, message, operator, stackStartFn) {
-  const argsLen = arguments.length;
+function fail(message) {
+  // TODO(@jasnell): When the Error.isError primordial is available, use it.
+  if (Error.isError(message)) throw message;  // eslint-disable-line node-core/prefer-primordials
```

Comment:
If we still want to rethrow non-errors having `Error.prototype` in their prototype chain, `internalUtil.isError` would be preferable.
```suggestion
  if (isError(message)) throw message;
```